webapp/src/components/training/Training.vue

```diff
@@ -38,22 +38,42 @@ import Finished from '@/components/training/Finished.vue'
 import Data from '@/components/data/Data.vue'
 
 const router = useRouter()
+const route = useRoute()
 const trainingStore = useTrainingStore()
 const tasksStore = useTasksStore()
 
 // task ID given by the route
 interface Props { id: TaskID }
 const props = defineProps<Props>()
 
+function setupTrainingStore() {
+  trainingStore.setTask(route.params.id as string) // more reliable than props.id
+  trainingStore.setStep(1)
+}
+// Init the task once the taskStore has been loaded successfully
+// If it is not we redirect to the task list
 const task = computed(() => {
-  const task = tasksStore.tasks.get(props.id)
-  if (task === undefined) {
-    router.replace({ name: 'not-found' })
-    return
+  if (tasksStore.loadedSuccessfully) {
+    return tasksStore.tasks.get(props.id)
+  }
+  // Redirect to the task list if not loaded yet
+  // This happens when refreshing the page, every task are reset when fetched
+  if (route.name !== 'task-list') {
+    router.replace({ name: 'task-list' })
+  }
+  return undefined
+})
+
+// Addresses the case when users enter a url manually
+// Force the training store to synch with the task specified in the url
+watch(() => route.fullPath, () => {
```

Comment:
hum, from a quick look at the doc, there seems to be [a nicer way to react to route changes](https://router.vuejs.org/guide/essentials/dynamic-matching.html#Reacting-to-Params-Changes)

Reply:
Well, I just tried it and I can't get onBeforeRouteUpdate to trigger so I'll leave it as it is. There are also some technical differences between the two that I don't have time to test.